scripts/eval_attribution.py

```
@@ -131,17 +144,55 @@ def searched_score(engine: UciEngine, nodes: int, white_stm: bool) -> int:
+        searched_nodes = re.search(r"\bnodes (\d+)", score_lines[-1])
+        if searched_nodes is None:
+            raise RuntimeError("final search score did not report its node count")
+        if int(searched_nodes.group(1)) < nodes:
+            raise RuntimeError(
+                f"final exact score was reported after {searched_nodes.group(1)} nodes; "
+                f"expected at least {nodes}"
+            )
+    if depth is not None:
+        searched_depth = re.search(r"\bdepth (\d+)", score_lines[-1])
+        if searched_depth is None:
+            raise RuntimeError("final search score did not report its depth")
```

Ruff (0.16.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)